web/src/input_pill.ts

```diff
@@ -243,6 +244,45 @@ export function create<T>(opts: InputPillCreateOptions<T>): InputPillContainer<T
             return undefined;
         },
 
+        // This is only used for search, and maybe shouldn't live here?
+        removeUserPill(user_container: HTMLElement, user_id: number) {
```

Comment:
I'd love some help with this commit. It's functional, but the code feels messy in several ways and I think I should be implementing it differently but I'm not sure what would be better.

Here are the concerns I have:

(1) The way this code reads feels a bit clunky (several `for` loops finding indexes of things) but I'm mirroring this off of `removePill` and I think it might be needed to be able to reach into the `store` to delete the pill?

(2) I feel weird putting this code in `input_pill` because it's very specific to search and not generic at all, but it does seem to need `store` to be able to edit the pill content. Would it be better to put this code elsewhere and have the `store` less encapsulated? Or is it better to just have this function here?

(3) ` const user_pill_container: InputPillItem<SearchUserPill> = store.pills[container_idx].item;` is having type error issues because the `store` treats the pills as template types. This feels related to the issues above, but I'm not sure what a good way of addressing it is.

Reply:
This seems readable enough to me. I'd consider just replacing the comment just above `removeUserPill` with something like:

> // TODO: This function is only used for the search input supporting multiple user pills within an individual top-level pill. Ideally, we'd encapsulate it in a subclass used only for search so that this code can be part of `search_pill.ts`.

Reply:
Okay, I'll add that comment and leave it there for now. Thanks!